classes/ActionScheduler_wpPostStore.php

```diff
@@ -638,4 +749,83 @@ public function init() {
 		$taxonomy_registrar = new ActionScheduler_wpPostStore_TaxonomyRegistrar();
 		$taxonomy_registrar->register();
 	}
+
+	/**
+	 * Get the last time the action was attempted.
+	 *
+	 * The time should be given in GMT.
+	 *
+	 * @param string $action_id
+	 *
+	 * @return DateTime
+	 * @throws InvalidArgumentException When the action cannot be found.
+	 */
+	public function get_last_attempt( $action_id ) {
+		$post = $this->get_valid_post_object( $action_id );
+
+		return as_get_datetime_object( $post->post_modified_gmt );
+	}
+
+	/**
+	 * Get the last time the action was attempted.
+	 *
+	 * The time should be given in the local time of the site.
+	 *
+	 * @param string $action_id
+	 *
+	 * @return DateTime
+	 * @throws InvalidArgumentException When the action cannot be found.
+	 */
+	public function get_last_attempt_local( $action_id ) {
+		$post = $this->get_valid_post_object( $action_id );
+
+		return as_get_datetime_object( $post->post_modified, get_option( 'gmt_offset', '0' ) );
```

Comment:
`ActionScheduler_TimezoneHelper::get_local_timezone()`, and by extension, `$this->get_local_timezone()` returns a `DateTimeZone` object, while we need a string representation of the timezone.

I think it would be worthwhile to extract the portion of `ActionScheduler_TimezoneHelper::get_local_timezone()` that determines the timezone string into a separate method, so I'll go that route.

Reply:
> we need a string representation of the timezone.

That wouldn't be needed if this method is refactored to:

```
$last_attempt_date = $this->get_last_attempt();
$last_attempt_date->setTimezone( ActionScheduler_TimezoneHelper::get_local_timezone() );
return $last_attempt_date;
```

Which is better anyway as it avoids duplicating the calls to `get_valid_post_object ()` to `as_get_datetime_object()` in both methods and instead _only_ applies logic to change what is different between this method and `get_last_attempt()` - the timezone.

Reply:
A much better option indeed, thanks! I'll use that instead.